policy-controller/k8s/index/src/outbound/index/http.rs

```diff
@@ -0,0 +1,310 @@
+                }
+
+                Some(timeout)
+            });
+
+    Ok(OutboundRouteRule {
+        matches,
+        backends,
+        request_timeout,
+        backend_request_timeout,
+        filters,
+    })
+}
+
+fn convert_gateway_rule(
+    ns: &str,
+    rule: gateway::HttpRouteRule,
+    cluster: &ClusterInfo,
+    service_info: &HashMap<ServiceRef, ServiceInfo>,
+) -> Result<OutboundRouteRule<HttpRouteMatch>> {
+    let matches = rule
+        .matches
+        .into_iter()
+        .flatten()
+        .map(routes::http::try_match)
+        .collect::<Result<_>>()?;
+
+    let backends = rule
+        .backend_refs
+        .into_iter()
+        .flatten()
+        .filter_map(|b| convert_backend(ns, b, cluster, service_info))
+        .collect();
+
+    let filters = rule
+        .filters
+        .into_iter()
+        .flatten()
+        .map(convert_gateway_filter)
+        .collect::<Result<_>>()?;
+
+    Ok(OutboundRouteRule {
+        matches,
+        backends,
+        request_timeout: None,
+        backend_request_timeout: None,
+        filters,
+    })
+}
+
+pub(crate) fn convert_backend<BackendRef: Into<gateway::HttpBackendRef>>(
+    ns: &str,
+    backend: BackendRef,
+    cluster: &ClusterInfo,
+    services: &HashMap<ServiceRef, ServiceInfo>,
+) -> Option<Backend> {
+    let backend = backend.into();
+    let filters = backend.filters;
+    let backend = backend.backend_ref?;
+    if !is_backend_service(&backend.inner) {
+        return Some(Backend::Invalid {
+            weight: backend.weight.unwrap_or(1).into(),
+            message: format!(
+                "unsupported backend type {group} {kind}",
+                group = backend.inner.group.as_deref().unwrap_or("core"),
+                kind = backend.inner.kind.as_deref().unwrap_or("<empty>"),
+            ),
+        });
+    }
+
+    let name = backend.inner.name;
+    let weight = backend.weight.unwrap_or(1);
+
+    // The gateway API dictates:
+    //
+    // Port is required when the referent is a Kubernetes Service.
+    let port = match backend
+        .inner
+        .port
+        .and_then(|p| NonZeroU16::try_from(p).ok())
+    {
+        Some(port) => port,
+        None => {
+            return Some(Backend::Invalid {
+                weight: weight.into(),
+                message: format!("missing port for backend Service {name}"),
+            })
+        }
+    };
+    let service_ref = ServiceRef {
+        name: name.clone(),
+        namespace: backend.inner.namespace.unwrap_or_else(|| ns.to_string()),
+    };
+
+    let filters = match filters
+        .into_iter()
+        .flatten()
+        .map(convert_gateway_filter)
+        .collect::<Result<_>>()
+    {
+        Ok(filters) => filters,
+        Err(error) => {
+            return Some(Backend::Invalid {
+                weight: backend.weight.unwrap_or(1).into(),
+                message: format!("unsupported backend filter: {error}", error = error),
```

Comment:
nitpicky, we could probably remove the `error = error` part
```suggestion
                message: format!("unsupported backend filter: {error}"),
```